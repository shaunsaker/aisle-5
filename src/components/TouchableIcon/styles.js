import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  container: {},
  icon: {
    fontSize: 30,
    color: styleConstants.colors.white,
    transform: [
      {
        translateX: -4,
      },
    ],
  },
});

export default styles;
