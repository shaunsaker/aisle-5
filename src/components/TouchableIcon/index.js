import React from 'react';
import PropTypes from 'prop-types';
import Icon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';

import Touchable from '../Touchable';

const propTypes = {
  iconName: PropTypes.string,
  handlePress: PropTypes.func,
};

const defaultProps = {};

const TouchableIcon = ({ iconName, handlePress }) => {
  return (
    <Touchable onPress={handlePress} style={styles.container}>
      <Icon name={iconName} style={styles.icon} />
    </Touchable>
  );
};

TouchableIcon.propTypes = propTypes;
TouchableIcon.defaultProps = defaultProps;

export default TouchableIcon;
