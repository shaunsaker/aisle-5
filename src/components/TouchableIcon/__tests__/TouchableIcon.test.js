import React from 'react';
import renderer from 'react-test-renderer';

import TouchableIcon from '..';

describe('TouchableIcon', () => {
  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<TouchableIcon />);

      expect(component).toMatchSnapshot();
    });
  });
});
