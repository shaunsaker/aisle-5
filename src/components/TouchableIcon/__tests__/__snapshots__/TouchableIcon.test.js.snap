// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TouchableIcon renders renders with minimum required props 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Object {
          "color": "#EDE8CA",
          "fontSize": 30,
          "transform": Array [
            Object {
              "translateX": -4,
            },
          ],
        },
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    
  </Text>
</View>
`;
